scrape/target.go

```diff
@@ -414,6 +419,23 @@ func PopulateLabels(lset labels.Labels, cfg *config.ScrapeConfig) (res, orig lab
 		lb.Set(model.AddressLabel, addr)
 	}
 
+	if noDefaultPort {
+		// If it's an address with a trailing default port and the
+		// no-default-scrape-port flag is present, remove the port.
+		switch port {
+		case "80":
+			if scheme == "http" {
+				lb.Set(model.AddressLabel, host)
+			}
+		case "443":
+			if scheme == "https" {
+				lb.Set(model.AddressLabel, host)
+			}
+		case "":
+			lb.Set(model.AddressLabel, addr2)
```

Comment:
This does not seem needed to me.

Reply:
Yeah, I don't think so. Might have been left over from when I moved this logic.